eclipse-che/che-operator
# Add fields for configuring project clone container to CheCluster CRD

### What does this PR do?
Adds fields to `checluster.spec.devEnvironments`:
```yaml
spec:
  devEnvironments:
    projectCloneContainer:
      name: <unused name field from container struct>
      image: <image used for project clone>
      imagePullPolicy: <image pull policy for project clone>
      env:
      - name: <env var name>
        value: <env var value
      resources:
        limits:
          cpu: <cpuLimit>
          memory: <memoryLimit>
        request:
          cpu: <cpuRequest>
          memory: <memoryRequest>
```
The `name` field is left over from reusing the existing Container struct as described in https://github.com/eclipse/che/issues/22036 and is consistent with `.spec.devEnvironments.gatewayContainer` added in https://github.com/eclipse-che/che-operator/pull/1639.

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
Project clone part of https://github.com/eclipse/che/issues/22036

### How to test this PR?
Deploy changes (including updated CRDs) and create a CheCluster with spec 
```yaml
spec:
  devEnvironments:
    projectCloneContainer:
      image: test-image
      imagePullPolicy: IfNotPresent
      env: 
      - name: test-env
        value: test-val
      resources:
        limits:
          memory: 1234Mi
          cpu: 1234m
        request:
          memory: 123Mi
          cpu: 123m
```
and verify that 
1. Fields are propagated into the DevWorkspaceOperatorConfig created by Che (the image used in the sample above will cause the workspace to fail to start)
2. When starting a workspace, fields defined in the CheCluster are used for the project clone init container
3. Editing/removing fields removes them from the DevWorkspaceOperatorConfig
### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
